packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -82,9 +86,13 @@
                 DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS,
                 true);
       }
+      MediaSource mediaSource;
+      if (isHls) {
```

Comment:
couldn't we eliminate this parameter by checking the URL for m3u8 instead?

Reply:
True. Thats what I thought first. But then there is a possibility that a http server can return the correct headers and data without actually having the right extension.
ex:
Given the file,
`https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8`
I can have a nodejs proxy server which exposes the route,
`https://localhost:3000/livestream` and the video will still work even though it does not have the right extension because it would send the right headers.

I don't know if the the HLS spec states that the file has to end with a `.m3u8` extension.
Also browsers try to detect the mime type on their own using headers and file metadata. So you can have the extension `.jpg` but return header `txt/html` and it would still render html.

Reply:
Also this is the mimetype.

.m3u8 | vnd.apple.mpegURL or application/x-mpegURL
-- | --
.ts | video/MP2T

If we can parse the header and get the mimetype that would make it easier. Don't know if exoplayer2 supports it.

Reply:
Also I've used react-native-video before and expo sdk and they do it automatically. Looking how they solve it using exoplayer2.
This is how they do it.
https://github.com/react-native-community/react-native-video/blob/master/android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java#L267

So I guess this will be ideal approach to solve it without passing extra optional parameters.
Will see if I can try it.